src/lighteval/data.py

```diff
@@ -204,6 +212,29 @@ def _sorting_criteria(self, request: LoglikelihoodSingleTokenRequest) -> int:
 
 
 class GenerativeTaskDataset(DynamicBatchDataset):
+    def init_split_limits(self, num_dataset_splits):
+        if num_dataset_splits is not None:
+            hlog_warn(
+                "You cannot select the number of dataset splits for a generative evaluation at the moment. Automatically inferring."
+            )
+
+        all_sorting_criterion = [self._sorting_criteria(self.sorted_data[0])[:2]]
```

Comment:
it's not very clear why we only take the first two element from the sorting criteria

Reply:
first 2 elements are whether to use logits and the eos token, and the latter is the item length (see [here](https://github.com/huggingface/lighteval/blob/5db69e8feae83d3b9de8590dad8fec4cab6514ba/src/lighteval/data.py#L251) ) - do you want me to add some doc?

Reply:
Adding some doc could be helpful yes !

Reply:
tell me if that's clearer now :)